pennylane/devices/capabilities.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Defines the DeviceCapabilities class, and tools to load it from a TOML file.
+"""
+
+import sys
+from dataclasses import dataclass, field, replace
+from enum import Enum
+from itertools import repeat
+
+if sys.version_info >= (3, 11):
+    import tomllib as toml
+else:
+    import tomli as toml
```

Comment:
Using `tomli` instead of `tomlkit` because `tomli` is already a dependency of PennyLane while `tomlkit` is not.

Reply:
@mlxd is the reason we chose `tomlkit`.  I'm fine with this, but I'll let him have the final say.